app/models/solidus_stripe/payment_method.rb

```diff
@@ -163,13 +163,18 @@ def payment_profiles_supported?
       false
     end
 
-    def stripe_dashboard_url(payment)
-      path_prefix = '/test' if preferred_test_mode
+    # Fetches the payment intent when available, falls back on the setup intent associated to the order.
+    def self.intent_id_for_payment(payment)
+      return unless payment
 
-      intent_id =
-        payment.transaction_id ||
+      payment.transaction_id || (
```

Comment:
The transaction_id will be populated in both cases but with SetupIntent, after the payment step and before confirming the order, the payment does not have a PaymentIntent nor a transaction_id yet. If we need to retrieve the intent at that point (eg. to show a link to stripe in the admin for that incomplete order), we have no way for doing that. 

That said, we know that `SolidusStripe::PaymentIntent.where(order: payment.order, payment_method: payment.payment_method)&.pick(:stripe_payment_intent_id)` is probably useless because it should always be the same as the `transaction_id`, but maybe there are some edge cases we are not considering and for now it might have sense to keep it. Open to alternatives!